docs/clients/promtail/stages/tenant.md

```diff
@@ -0,0 +1,37 @@
+# `tenant` stage
+
+The tenant stage is an action stage that sets the tenant ID for the log entry
+picking it from a field in the extracted data map. If the field is missing, the
+default promtail client [`tenant_id`](../configuration.md#client_config) will
+be used.
+
+
+## Schema
+
+```yaml
+tenant:
+  # Name from extracted data to whose value should be set as tenant ID
+  source: <string>
+```
+
+### Examples
+
+For the given pipeline:
+
+```yaml
+- json:
```

Comment:
I think a match stage would be a good example too here.

```yaml
- match:
    selector: {namespace="team-a"}
    stages:
     - template:
         source: tenantid
         template: 'XXXX'
    - tenant: tenantid
        source: tenantid
```

This would show how to configure promtail per namespace per team.

Reply:
Thanks for your feedback. Would you mind checking the updated doc and eventually help me improving it, please?

Reply:
looks good !